extension/intents/navigation/navigation.js

```diff
@@ -51,19 +52,31 @@ intentRunner.registerIntent({
 intentRunner.registerIntent({
   name: "navigation.bangSearch",
   async run(context) {
-    const service = context.slots.service || context.parameters.service;
+    let service = context.slots.service || context.parameters.service;
+    let tab = undefined;
+
+    if (service === undefined) {
+      service = await serviceList.detectServiceFromActiveTab(metadata.search);
+      tab = await context.activeTab();
```

Comment:
Note this code currently fails if you search on a site that isn't supported. This is what I get:

```
Error in intent navigation.bangSearch : TypeError: can't access property "toLowerCase", name is null ddgBangServiceName@moz-extension://52ad6e95-d3ed-b243-adfd-fce5310dae53/background/serviceList.js:17:16
ddgBangSearchUrl@moz-extension://52ad6e95-d3ed-b243-adfd-fce5310dae53/searching.js:56:28
run@moz-extension://52ad6e95-d3ed-b243-adfd-fce5310dae53/intents/navigation/navigation.js:63:35
async*runIntent@moz-extension://52ad6e95-d3ed-b243-adfd-fce5310dae53/background/intentRunner.js:278:18
runUtterance@moz-extension://52ad6e95-d3ed-b243-adfd-fce5310dae53/background/intentRunner.js:255:10
@moz-extension://52ad6e95-d3ed-b243-adfd-fce5310dae53/background/main.js:40:25
```

Probably instead of giving an error this should open up a regular search with `site:${(new URL(tab.url)).origin}` added to the query.